internal/alpini/router/alpini-router-base/src/main/java/com/linkedin/alpini/router/ScatterGatherRequestHandlerImpl.java

```diff
@@ -679,7 +679,7 @@ private void prepareRetry(AsyncFuture<HRS> retryFuture) {
         boolean isRequestRetriable = isRequestRetriable(path, role, retryStatus);
 
         // Indicate the current request is a retry request
-        path.setRetryRequest();
+        path.setRetryRequest(retryFuture.isSuccess() ? null : retryFuture.getCause());
```

Comment:
I think we can use `retryStatus` directly here:
```
path.setRetryRequest(retryStatus)
```

Reply:
Do you know what exactly the type retryStatus is it is saying HRS but that's generics I believe.